clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -106,7 +107,7 @@ public AbstractConfig(ConfigDef definition, Map<?, ?> originals,  Map<String, ?>
 
         this.originals = resolveConfigVariables(configProviderProps, (Map<String, Object>) originals);
         this.values = definition.parse(this.originals);
-        this.used = Collections.synchronizedSet(new HashSet<>());
+        this.used = new CopyOnWriteArraySet<>();
```

Comment:
How about a ConcurrentHashSet instead? Not a big fan of CopyOnWriteArraySet.

Reply:
Yeah, I normally don't like them very much either but I thought the number of operations on it would be small and it'd be OK. Given that this used map is updated on every use, maybe that assumption could end up being wrong in some use of AbstractConfig. I'll switch it to ConcurrentHashSet.

Reply:
Done